vignettes/adrs_basic.Rmd

```diff
@@ -0,0 +1,873 @@
+)
+```
+
+### Derive `AVALC` and `AVAL`
+
+Here we populate `AVALC` and create the numeric version as `AVAL`
+(ordered from best to worst response). This ordering is already covered
+within our RECIST v1.1 parameter derivation functions, and so changing
+`AVAL` here would not change the result of those derivations.
+
+```{r}
+adrs <- adrs %>%
+  mutate(
+    AVALC = RSSTRESC,
+    AVAL = aval_resp(AVALC)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, RSSTRESC, AVALC, AVAL)
+)
+```
+
+### Derive `ANL01FL` {#anl01fl}
+
+When deriving `ANL01FL` this is an opportunity to exclude any records
+that should not contribute to any downstream parameter derivations. In
+the below example this includes only selecting valid assessments and
+those occurring on or after randomization date. If there is more than
+one assessment at a date, the worst one is flagged.
+
+```{r}
+adrs <- adrs %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = exprs(STUDYID, USUBJID, ADT),
+      order = exprs(AVAL, RSSEQ),
+      new_var = ANL01FL,
+      mode = "last"
+    ),
+    filter = !is.na(AVAL) & ADT >= RANDDT
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, AVALC, ADT, RANDDT, ANL01FL)
+)
+```
+
+Here is an alternative example where those records occurring after new
+anti-cancer therapy are additionally excluded (where `NACTDT` would be
+pre-derived as first date of new anti-cancer therapy. See `{admiralonco}` [Creating and Using New Anti-Cancer Start Date](nactdt.html) for deriving this variable).
+
+```{r, eval=FALSE}
+adrs <- adrs %>%
+  mutate(
+    ANL01FL = case_when(
+      !is.na(AVAL) & ADT >= RANDDT & ADT < NACTDT ~ "Y",
+      TRUE ~ NA_character_
+    )
+  )
+```
+
+Note here that we don't filter out records after first `PD` at this
+stage, as that is specifically catered for in the `{admiralonco}`
+parameter derivation functions in the below steps, via `source_pd`
+arguments. 
+
+### Derive `ANL02FL` {#anl02fl}
+
+However, if you prefer not to rely on `source_pd` arguments,
+then the user is free to filter out records after first `PD` at this
+stage in a similar way via a `ANLzzFL` flag, and then you could leave
+`source_pd` as null in all downstream parameter derivation function
+calls. So, for example the user could create `ANL02FL` flag to subset
+the post-baseline response data up to and including first reported
+progressive disease. This would be an alternative and transparent method
+to the use of `source_pd` argument approach to create ADRS parameters
+below. Using `{admiral}` function `admiral::derive_var_relative_flag()`
+we could create `ANL02FL` as below.
+
+```{r, eval=FALSE}
+adrs <- adrs %>%
+  derive_var_relative_flag(
+    by_vars = exprs(USUBJID),
+    order = exprs(ADT, AVISITN),
```

Comment:
should we use same as new adrs vignette/template here? i.e.:
```
    by_vars = exprs(STUDYID, USUBJID),
    order = exprs(ADT, RSSEQ),
```

Reply:
Yes, updated

Reply:
i don't see this update made?

Reply:
Sorry, I updated `by_vars` but missed `order`. It is updated now.